go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -588,7 +589,36 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		}
 	}
 
-	ops, err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitchOps(bnc.nbClient, nil, ls, lsp)
+	lsps := []*nbdb.LogicalSwitchPort{lsp}
+	if kubevirt.IsPodOwnedByVirtualMachine(pod) && isAllowedForMigration(bnc.IsSecondary(), bnc.IsPrimaryNetwork(), bnc.isLayer2Interconnect()) {
```

Comment:
I don't understand why you're checking if `isLayer2Interconnect`. 

We should be able to benefit from better migrations in non-IC scenarios as well.

Reply:
we want this for localnet as well.

Is there a reason not to have this feature for localnet ?

Reply:
@maiqueb can we do localnet at follow up PR ?

Reply:
why ? what's the extra effort ?

Reply:
Like adding more e2e tests maybe

Reply:
If we add those, the only difference is having a table and parametrize the topology. 

FWIW, I'm OK with adding e2e tests in a follow up PR introducing a "maximal downtime" metric that must be met during migration.

Reply:
Well first we need to add the basic live migration ones where we check we don't break the tcp connection.

Reply:
But don't we have tests for those in the kubevirt lanes ? 

We need to:
- ensure we don't break live migration
- ensure the live migration performs better with this PR than it did before

The former is necessary. And we have e2e tests for that in the code base.

The latter is IMHO not required right now: we can check it manually, and add those tests (that require introducing a "maximal downtime" metric) to a follow up PR.

Reply:
We don't have break live migration e2e test at code base since we are using passt there.

Reply:
ACK, so.. 
- Removing `bnc.isLayer2Interconnect()`
- Allowing `bnc.IsPrimaryNetwork()`
- Adding `bnc.TopologyType() != layer3`

Reply:
> We don't have break live migration e2e test at code base since we are using passt there.

So let's change that.

Reply:
we first need the managedTap PR though.

Reply:
OK, so let's bring that in ASAP.